File: 1441 build-an-array-with-stack-operations/solution.go
func buildArray(target []int, n int) []string {
    var ans []string 
    idx:=0
    for i:=1;i<=n && idx<len(target);i++ {
        ans = append(ans,"Push")
        if target[idx]==i {
            idx++
        } else {
            ans = append(ans,"Pop")
        }
    }
    return ans
}

File: 1441 build-an-array-with-stack-operations/solution.java
class Solution {
    public List<String> buildArray(int[] target, int n) {
        List<String> ans = new ArrayList<>();
        int idx=0;
        for(int i=1;i<=n && idx<target.length;i++) {
            ans.add("Push");
            if(target[idx]==i){
                idx++;
            }
            else {
                ans.add("Pop");
            }
        }
        return ans;
    }
}
